server/app/views/store_product.py

```
@@ -0,0 +1,314 @@
+
+        return PaginatedStoreProducts(
+            data=store_products,
+            total=total,
+            page=(skip // limit) + 1,
+            page_size=limit,
+            total_pages=total_pages,
+        )
+
+    @router.get("/{upc}", response_model=StoreProduct, operation_id="getStoreProduct")
+    async def get_store_product(
+        self, upc: str, repo: StoreProductRepository = Depends(store_product_repository)
```

Ruff (0.11.9)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)